doc/source/templates/README.md

```diff
@@ -16,79 +16,129 @@ Coming soon...
-    | Number of Nodes | 1 | > 1 |
-    | Dataset size | Subset (of partitions/labels/rows) | Full example dataset |
-    | Model size | Pruned/mini version of the model | Full model |
-    | Runtime | 30-60s | Up to ~5-10 minutes |
+    NOTE: The need for a second test copy is temporary. Only one notebook will be needed
+    from 2.5 onward.
 
-    When you specify the template's compute config, see `doc/source/templates/configs` for defaults.
+    **Label all test-specific code with the `remove-cell` Jupyter notebook tag.**
+
+    **Put this test copy in `doc/source/templates/tests/test-copy.ipynb`.**
 
-3. Add a nightly release test for the template in `release/release_tests.yaml`.
+3. Add a smoke-test version of the template to run in pre-merge CI in `doc/BUILD` under the templates section.
+
+    Set the `SMOKE_TEST` environment variable, which should be used in your template to
+    set certain smoke test parameters (like limiting dataset size).
+
+    **Make sure that you tag the test with `"gpu"` if required, and any other tags
+    needed for special dependencies.
+
+    ```python
+    py_test_run_all_notebooks(
+        size = "large",
+        include = ["source/templates/tests/batch_inference.ipynb"],
+        exclude = [],
+        data = ["//doc:workspace_templates"],
+        tags = ["exclusive", "team:ml", "ray_air", "gpu"],
+        env = {"SMOKE_TEST": "1"},
+    )
+    ```
+
+4. Add a release test for the template in `release/release_tests.yaml` (for both AWS and GCE).
+
+    **Use the `release_test_cluster_env.yaml` and `*_release_test.yaml` files for cluster env / compute configs.**
+    These contain placeholders for regions and cloud ids that our CI infra will fill in.
 
     ```yaml
     - name: workspace_template_small_02_many_model_training
       group: Workspace templates
-      working_dir: workspace_templates/02_many_model_training
+      working_dir: workspace_templates/tests
       python: "3.9"
-      frequency: nightly
+      frequency: nightly-3x
       team: ml
       cluster:
         cluster_env: ../configs/release_test_cluster_env.yaml
-        cluster_compute: ../configs/compute/cpu/aws_small.yaml
+        cluster_compute: ../configs/compute/cpu/aws_release_test.yaml
+
+      variations:
+          - __suffix__: aws
+          - __suffix__: gce
+            env: gce
+            frequency: manual
+            cluster:
+              cluster_env: ../configs/release_test_cluster_env.yaml
+              cluster_compute: ../configs/compute/gpu/gce_release_test.yaml
 
     run:
       timeout: 300
-      script: pip install -U -r requirements.txt
-        && jupyter nbconvert --TagRemovePreprocessor.remove_input_tags='large'
-        --to script --output _test many_model_training.ipynb && ipython _test.py
+      script: jupyter nbconvert --to script --output _test many_model_training.ipynb && ipython _test.py
+    ```
+
+5. (Anyscale Only) To make this template show up as an Anyscale Workspace Template:
```

Comment:
Move this step to the product repo?

Reply:
Sounds good. I'll centralize the full set of steps on a wiki page that can link both the readmes.